[v1.4][W14-B3] HR+

tanhengyeow opened the pull request:
@mhq199657 @kexiaowen @Ang-YC

mhq199657 reviewed (approved):
Looks good to me! 👍

tanhengyeow commented:
**End of v1.0 milestone**

kowshik-sundararajan commented on README.adoc:
```diff
@@ -1,11 +1,11 @@
 = Address Book (Level 4)
```
Remember to update the name of your app - it is not an Address Book anymore!

kowshik-sundararajan commented on docs/DeveloperGuide.adoc:
```diff
@@ -10,7 +10,7 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level4/tree/master
+:repoURL: https://github.com/CS2103JAN2018-W14-B3/main/tree/master
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Team name should be updated as well.

kowshik-sundararajan reviewed (commented):
Overall, good work on updating user stories, use cases, other NFRs and JUnit tests! 👍 

Some general feedback:
* Good to see that you have created milestones. Please remember to add deadlines to the `v1.1` - `v1.5` milestones.

* Good work on setting up the issue tracker in your repo. As you progress, use more tags to filter between NFRs, enhancements, etc.  

* I see that a couple of individual PRs that didn't pass Travis have been merged in your repo - in practice, this shouldn't be acceptable. By merging pull requests that do not pass Travis, you will find it difficult to figure out why the main build fails.

*Do not close this PR!*

tanhengyeow commented:
Hello @kowshik-sundararajan 

Thanks for the review! We'll update the changes to the docs accordingly 😄 

All PRs did pass both Travis and AppVeyor builds actually before we merge them. One failed due to decreased coverage and some were pending due to some anomalies in Coveralls. 

The Coverage build was passing when we checked it manually but the PR check was not updating in Github so we just merged the PRs after confirming that the Travis/AppVeyor builds passed. 

I did a reset and the Coveralls check is working fine now!

kowshik-sundararajan commented:
Sweet @tanhengyeow!
I should have taken a closer look 🙂

mhq199657 commented:
Dear @kowshik-sundararajan The Travis Build fails because it is not running on our latest code. We have reflected this problem to the TA during yesterday's tutorial.

mhq199657 commented:
Dear @kowshik-sundararajan Travis still does not run on our code. Could you help us fix this problem? The `Travis CI build failed` notification is annoying especially when our code actually does not fail.

kowshik-sundararajan commented on docs/UserGuide.adoc:
```diff
@@ -11,7 +11,7 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level4
+:repoURL: https://github.com/CS2103JAN2018-W14-B3/main
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Please update this to your team number: `W14-Bx`.

kowshik-sundararajan commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,16 +890,212 @@ Use case ends.
+*MSS*
+
+1.  User requests to add a new person in the list
+2.  AddressBook adds the new person into the address book.
```
Remember to update all references to `AddressBook` to your application's name!

kowshik-sundararajan commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,16 +890,212 @@ Use case ends.
 Use case resumes at step 2.
 
-_{More to be added}_
+=== Use case: Add person
```
You can be more specific for all the use cases - replacing person with student.

kowshik-sundararajan commented on docs/UserGuide.adoc:
```diff
@@ -158,6 +195,23 @@ Selects the 2nd person in the address book.
+Examples:
+
+* `rate 1 t/4.5 c/3 p/4 e/3.5` +
+Rates the 1st person to have technical skills score of 4.5, communication skills score of 3, problem solving skills of 4, experience score of 3.5.
```
Since your description says `rates the student`, ensure that it is consistent throughout all the documentation.

kowshik-sundararajan reviewed (commented):
@tanhengyeow  @mhq199657  @kexiaowen  @Ang-YC
Please indicate which features appear in which milestone. Also If there is something that is left for future work, need to indicate it as coming in v2.0

I would want you to mention the following in your **Developer Guide**:

* For each team member what are the features (major and minor) you are proposing?
* Within the scope of the project, how does it fit in (just a one or 2 line summary)
Please do this by end of day Monday 19 March.

General feedback: 
* Include features and their description that are planned to be implemented in `v2.0` in the User Guide.
* Remember to replace all references to AddressBook 😛

kowshik-sundararajan commented:
As a favour, could you add a comment every weekend to summarize what are the changes that have been made? This would make it easier for me to review the changes.

About Travis, let me look into it and see what can be done.

tanhengyeow commented:
Hi @kowshik-sundararajan 

Thanks for the review!

Here is the summary of work done for our v1.1 milestone:
1) Added an optional `resume` field to each student
2) Added a `rate` command and `ratings` field to each student
3) Improved the `find` command to extend exact keyword search to all fields
4) Added a `interview` command and `interview` field to each student
5) Updated documentation for v1.1

Each PR has a write up that explains the changes in depth :smile:

kowshik-sundararajan commented:
Great! Additionally, if you want me to review your code for quality checks, tag me in the PRs submitted to your repo.

Note for next week: Each member of the team should have at least 1 merged PR of their enhancement as `v1.2` release. (It doesn't have to be perfect code, but something that works).

kexiaowen commented:
**Summary of changes for `v1.2`**

Hongqiang: 
- Improve `filter` command to make it always filter the current view of the list instead of all the persons in HR+. 
- Support advanced filter predicate strings, in the form of e, a - b, c, d - f (single or interval connected with commas). Logical predicates separated with comma are ORed. 

Xiaowen:
- Add `overall score` for `rating` which is calculated as the average of technical skills score, communication skills score, problem solving skills score and experience scores.
- Add `deleteRating` command that deletes the rating of the person at the specified index.

Heng Yeow:
- Add a `Major` class and support adding major in the `add` command. Make this field editable as well.
- Improve `find` command by adding new predicate and predicate test for field `major`. Refactor methods in `FindCommandParser` to `FindUtil`.

Yee Chin:
- Revamp the interface to make it more intuitive and clean, allowing users to find the necessary information easily. (Details can be found in the PR created)

@kowshik-sundararajan

mhq199657 committed:
Make final declaration only be edited once

mhq199657 committed:
Support filter by rating using prefix r/

mhq199657 committed:
fix style

mhq199657 committed:
trigger codacy again

mhq199657 committed:
Add unit test for new predicate

mhq199657 committed:
Add new test for new methods in FilterUtil

mhq199657 committed:
Add new test for new methods in FilterCommandParser

mhq199657 committed:
Add new test for new methods in FilterCommand

mhq199657 committed:
Add new test for new methods in FilterCommandSystemTest

mhq199657 committed:
Fix style

tanhengyeow committed:
Merge pull request #53 from mhq199657/add-status-field

Add status field and status command

tanhengyeow committed:
Added GradePointAverage class

tanhengyeow committed:
Updated field GradePointAverage in relevant files

tanhengyeow committed:
Updated tests and relevant files related to tests

tanhengyeow committed:
Updated UG

kowshik-sundararajan reviewed (commented):
@tanhengyeow @mhq199657 @Ang-YC @kexiaowen 
Since I've reviewed the DG and your enhancements in the tutorial, there's not much to add on here.

For your UG, please specify which features have been implemented in `v1.0`, `v1.1`, `v1.2` and which features will be implemented in the remaining milestones.

kexiaowen committed:
Merge pull request #55 from mhq199657/improve-filter-command-ii

Improve filter command II

tanhengyeow committed:
Resolved merge conflict

kexiaowen committed:
Merge pull request #57 from tanhengyeow/v1.3-add-new-fields

Added field GradePointAverage

kexiaowen committed:
Add Job Applied Field to Person class

kexiaowen committed:
Update parsers and data

kexiaowen committed:
Update xml data

kexiaowen committed:
Update and add new test cases

kexiaowen committed:
Update UserGuide

kexiaowen committed:
Resolve merge conflicts

mhq199657 committed:
Fix resume startup and update UG

tanhengyeow committed:
Merge pull request #58 from kexiaowen/jobApplied

Add a new field JobApplied

tanhengyeow committed:
Remove unused method in FindCommand

tanhengyeow committed:
Added new method getAllPrefixes

tanhengyeow committed:
Added new methods for predicates

tanhengyeow committed:
Removed field ExpectedGraduationYear

tanhengyeow committed:
Removed files related to ExpectedGraduationYear predicate

tanhengyeow committed:
Added new predicates for fields

tanhengyeow committed:
Added a main predicate class for each field

tanhengyeow committed:
Added interface FieldPredicate

tanhengyeow committed:
Added main predicate class that manages all predicate field classes

tanhengyeow committed:
Refactored functionalities for FindCommand

tanhengyeow committed:
Updated tests

tanhengyeow committed:
Merge branch 'master' into v1.3-find-substring-prefix-suffix

tanhengyeow committed:
Improve code quality for StringUtil

tanhengyeow committed:
Updated UserGuide

tanhengyeow committed:
Updated DeveloperGuide

tanhengyeow committed:
Fixed typo

mhq199657 committed:
Fix test

mhq199657 committed:
Merge pull request #61 from tanhengyeow/v1.3-documentation

v1.3 documentation

mhq199657 committed:
Fix merge conflict

mhq199657 committed:
Fix style

mhq199657 committed:
Adhere to documentation standard

kexiaowen committed:
Add a command to sort students based on overall rating

kexiaowen committed:
Update parsers to accept new rate_sort command

kexiaowen committed:
Implement methods in Model component to support sort by overall rating

kexiaowen committed:
Add and update relevant tests

kexiaowen committed:
Update UserGuide

kexiaowen committed:
Fix codacy issue

mhq199657 committed:
Merge pull request #60 from tanhengyeow/v1.3-find-substring-prefix-suffix

Improved find command (wildcard search)

tanhengyeow committed:
Merge pull request #62 from mhq199657/fix-resume-invalid-on-startup

Fix resume invalid on startup

tanhengyeow committed:
Merge pull request #64 from kexiaowen/rating_sort

Add a rating_sort command to sort current list of students based on overall rating

mhq199657 committed:
fix sample data resume dependencies

mhq199657 committed:
fix startup issue

mhq199657 committed:
fix

tanhengyeow committed:
Merge pull request #65 from mhq199657/fix-startup-crash

Fix startup crash

Ang-YC committed:
Added profile image and comment field

Ang-YC committed:
Fixed code style

mhq199657 committed:
Merge branch 'master' into func-comments-photo

tanhengyeow committed:
Merge pull request #66 from Ang-YC/func-comments-photo

Added profile image and comment field

kexiaowen committed:
Rename commands to rating-delete and rating-sort

mhq199657 committed:
Support filtering by gpa

mhq199657 committed:
Support filtering by interviewDate

mhq199657 committed:
Fix styles

kexiaowen committed:
Add a new field University to Person class

kexiaowen committed:
Update xml files by adding in the new field university

kexiaowen committed:
Update test cases

kexiaowen committed:
Update User Guide

mhq199657 committed:
Fix tests

mhq199657 committed:
fix codacy

kexiaowen committed:
Fix typo in User Guide

kexiaowen committed:
Trigger codacy check again

tanhengyeow commented:
**Summary of work done for `v1.3`**

Hongqiang:
1) Added `status` field
2) Improve `filter` command to support multiple fields

Heng Yeow:
1) Added `Grade Point Average` field
2) Improve `find` command to support wildcard search

Xiaowen:
1) Added `Job Applied` field
2) Added `rating_sort` command to sort current list of students based on overall rating 

Yee Chin:
1) Added `Profile Image` and `Comments` field
2) Made user interface more responsive

@kowshik-sundararajan

kexiaowen committed:
Extend rating-sort command to a more generic sort command

kexiaowen committed:
Add and update test cases

tanhengyeow committed:
Merge pull request #67 from kexiaowen/University

Add a new field University

kexiaowen committed:
Merge pull request #68 from mhq199657/improve-filter-command-iii

Improve filter command III

kexiaowen committed:
Resolve merge conflict and update user guide

tanhengyeow committed:
Fixed typo in comments

Ang-YC committed:
Enhance unit tests of comment and profile image fields

tanhengyeow committed:
Merge pull request #71 from kexiaowen/sort

Extend current rating-sort to a generic sort command that supports sorting by name, gpa and overall rating

tanhengyeow committed:
Merge pull request #72 from Ang-YC/func-comments-photo-fix

Enhance unit tests of comment and profile image fields

tanhengyeow committed:
Merge branch 'master' into v1.4-find-fuzzy

tanhengyeow committed:
Refactored utility methods into FindResults

tanhengyeow committed:
Created singleton class to provide string utility functions for predicates and stores find results

tanhengyeow committed:
Added more prefixes in FindCommandParser

tanhengyeow committed:
Remove unnecessary tests in StringUtilTest

tanhengyeow committed:
Added new method to display find results

tanhengyeow committed:
Updated corresponding files to reflect new prefixes and to use comma separated delimiter now

tanhengyeow committed:
Updated exisiting predicates and added new predicates to perform fuzzy matching

mhq199657 committed:
Update documentation

mhq199657 committed:
Update documentation

mhq199657 committed:
Include specifics about resume

mhq199657 committed:
collate

kexiaowen committed:
Collate code and update documentation

mhq199657 committed:
collate tests

kexiaowen committed:
Correct typo in User Guide

mhq199657 committed:
collate

kexiaowen committed:
Merge pull request #73 from mhq199657/v1.4-documentation

V1.4 documentation

kexiaowen committed:
Merge branch 'master' into code-collate-kexiaowen

kexiaowen committed:
Merge pull request #74 from kexiaowen/code-collate-kexiaowen

Collate code and update documentation

kexiaowen committed:
Merge pull request #75 from mhq199657/v1.4-collation

V1.4 collation

tanhengyeow committed:
Updated predicate, command and system tests

tanhengyeow committed:
Merge branch 'master' into v1.4-find-fuzzy

tanhengyeow committed:
Fix checkstyle errors

tanhengyeow committed:
Collated code

tanhengyeow committed:
Trigger checks again

tanhengyeow committed:
Trigger checks again

tanhengyeow committed:
Updated UG

kexiaowen committed:
Merge pull request #78 from tanhengyeow/v1.4-find-fuzzy-textResults

Improved find command (fuzzy search) + text display of search results

kexiaowen committed:
Merge pull request #79 from tanhengyeow/v1.4-documentation

v1.4 documentation

Ang-YC committed:
Enhance and update User Interface

tanhengyeow committed:
Merge pull request #80 from Ang-YC/func-ui-v2

Enhance and update User Interface

Ang-YC committed:
Collate v1.4

tanhengyeow committed:
Merge pull request #81 from Ang-YC/collate-v1.4

Collate v1.4

tanhengyeow committed:
Added collated files

tanhengyeow committed:
Merge pull request #82 from tanhengyeow/v1.4-collate

Added collated files

mhq199657 committed:
Add persons

mhq199657 committed:
Supports copying in AddCommand

tanhengyeow committed:
Fixed some typos

mhq199657 committed:
Add tests

tanhengyeow committed:
Made statement clearer

tanhengyeow committed:
Removed unnecessary lines

tanhengyeow committed:
Merge pull request #83 from mhq199657/create-more-person

Add more Persons

mhq199657 committed:
Fix tests

tanhengyeow committed:
Merge pull request #84 from tanhengyeow/v1.5-documentation

Fixed typos to UG

Ang-YC committed:
😎 Added animated UI, radar chart and fixed minor bugs

Ang-YC committed:
😎 Added animated UI, radar chart and fixed minor bugs

mhq199657 committed:
Support profile image also

Ang-YC committed:
Cache everything for performance

Ang-YC committed:
Cache everything for performance

Ang-YC committed:
Increase profile image performance

Ang-YC committed:
Increase profile image performance

mhq199657 committed:
Add project portfolio

kexiaowen committed:
Added project portfolio page

kexiaowen committed:
Add Sort Feature in Developer Guide

kexiaowen committed:
Re-upload diagrams

kexiaowen committed:
Update user guide

kexiaowen committed:
Merge pull request #85 from mhq199657/copy-resume

Copy resume and profile image and add profile portfolio

kexiaowen committed:
Merge remote-tracking branch 'upstream/master' into v1.5-documentation

kexiaowen committed:
Merge pull request #112 from kexiaowen/v1.5-documentation

 Add Implementation of Sort Command in Developer Guide

kexiaowen committed:
Merge pull request #113 from kexiaowen/xw-project-portfolio

Add Project Portfolio Page

Ang-YC committed:
Added floating real time parser

Ang-YC committed:
Added floating real time parser

Ang-YC committed:
Reduce instances of Natty parser

Ang-YC committed:
Reduce instances of Natty parser

tanhengyeow committed:
Updated code admin stuff

tanhengyeow committed:
Merge branch 'master' into v1.5-code-admin

Ang-YC committed:
Added animated flag and log

Ang-YC committed:
Added animated flag and log

tanhengyeow commented:
**Summary of work done for `v1.4`**

Hongqiang:
1) Improved filter command to filter `interview dates` and `GPA`

Heng Yeow:
1) Improved `find` command to support fuzzy search matching
2) Displayed text results of `find` command in application console

Xiaowen:
1) Added `University` field
2) Supported sorting by different fields including gpa, overall rating and name

Yee Chin:
1) Enhanced unit tests of  `Profile Image` and `Comments` field
2) Enhanced and updated User Interface 

@kowshik-sundararajan

Ang-YC committed:
Added show command

Ang-YC committed:
Added show command

Ang-YC committed:
Replace some address book into HR+

Ang-YC committed:
Replace some address book into HR+

Ang-YC committed:
Updated something

Ang-YC committed:
Updated something

Ang-YC committed:
Merge branch 'func-ui-v3-conflict' into func-ui-v3

# Conflicts:
#	src/main/java/seedu/address/model/person/ProfileImage.java
#	src/main/java/seedu/address/model/person/Resume.java

Ang-YC committed:
Fix some codacy issue

kexiaowen committed:
Update user guide to resolve issues reported during peer-testing

kexiaowen committed:
Modify command format specification

kexiaowen committed:
Update documentation

kexiaowen committed:
Add collated code

kexiaowen committed:
Make sort field parameter case insensitive

kexiaowen committed:
Merge pull request #114 from Ang-YC/func-ui-v3

What's better than animated UI? 🙃

Ang-YC committed:
Added project portfolio

Ang-YC committed:
Merge pull request #115 from kexiaowen/v1.5rc

Fix issues and update documentation

Ang-YC committed:
Minor fix

tanhengyeow committed:
Updated UGDG and portfolio

tanhengyeow committed:
Merge pull request #116 from Ang-YC/docs-portfolio

Added project portfolio

tanhengyeow committed:
Merge branch 'master' into v1.5-code-admin

tanhengyeow committed:
Merge branch 'master' into v1.5-documentation

Ang-YC committed:
Quick fix for undo UI issue

tanhengyeow committed:
Merge pull request #117 from tanhengyeow/v1.5-code-admin

v1.5 code administrative stuff

Ang-YC committed:
Fix trailing whitespace

tanhengyeow committed:
Merge pull request #118 from tanhengyeow/v1.5-documentation

V1.5 documentation

tanhengyeow committed:
Merge pull request #119 from Ang-YC/quickfix-ui

Quick fix for undo UI issue

Ang-YC committed:
Updated UI on README

mhq199657 committed:
Merge pull request #120 from Ang-YC/docs-ui-v2

Updated UI on README

Ang-YC committed:
Revert list change UI update heuristic

Ang-YC committed:
Fix reference removal bug

Ang-YC committed:
Show resume button as disabled instead of hide

mhq199657 committed:
Merge pull request #121 from Ang-YC/bugfix-ui

Fix some UI bugs

kexiaowen committed:
Add rating-edit command to edit rating scores

kexiaowen committed:
Update tests

kexiaowen committed:
Update documentation

kexiaowen committed:
Add tests

tanhengyeow committed:
Removed AppVeyor

kexiaowen committed:
Merge pull request #122 from kexiaowen/rating-edit

Add rating-edit command

kexiaowen committed:
Merge pull request #123 from tanhengyeow/v1.5-admin

v1.5 admin

tanhengyeow committed:
Updated static methods to instance methods

Ang-YC committed:
Added ShowCommand tests

kexiaowen committed:
Minor update

Ang-YC committed:
Added UiUtil tests

kexiaowen committed:
Update collated code

kexiaowen committed:
Update README and AboutUs page

kexiaowen committed:
Merge pull request #124 from kexiaowen/v1.5-doc

Collate code

Ang-YC committed:
Added more WindowGui tests

Ang-YC committed:
Added ResumeGui tests

Ang-YC committed:
Removed PersonCard unused method

Ang-YC committed:
Added RealTimeParser tests

Ang-YC committed:
Added ShowCommandParser tests

Ang-YC committed:
Added Person tests

Ang-YC committed:
Added collated code

Ang-YC committed:
Fix Codacy analysis issue

Ang-YC committed:
Codacy sucks

Ang-YC committed:
Codacy again

tanhengyeow committed:
Merge pull request #125 from Ang-YC/tests-all

Added more tests

mhq199657 committed:
dg update

mhq199657 committed:
update

kexiaowen committed:
Fix rating-edit parameters range check

mhq199657 committed:
corrupted XML behaviour

kexiaowen committed:
Change equality check for person and update UG

mhq199657 committed:
Make person count to 20

kexiaowen committed:
Add Instructions for Manual Testing

kexiaowen committed:
Merge pull request #126 from mhq199657/master

Update DG

kexiaowen committed:
Merge pull request #127 from kexiaowen/rating-edit-fix

Fix rating-edit parameters range check

kexiaowen committed:
Minor update

kexiaowen committed:
Merge pull request #128 from kexiaowen/v1.5-UG

UGDG updates

Ang-YC committed:
Updated UG and DG

tanhengyeow committed:
Updated test cases

tanhengyeow committed:
Merge branch 'master' into v1.5-test-cases

tanhengyeow committed:
Trigger checks

tanhengyeow committed:
Trigger checks

tanhengyeow committed:
Merge pull request #130 from tanhengyeow/v1.5-test-cases

v1.5 test cases

Ang-YC committed:
Updated DG manual testing

tanhengyeow committed:
Merge pull request #129 from Ang-YC/docs-ug-dg

Updated UG and DG

tanhengyeow committed:
Merge pull request #131 from Ang-YC/docs-ug-dg-v2

Updated DG manual testing

mhq199657 committed:
Update manual testing

kexiaowen committed:
Modify UGDG

kexiaowen committed:
Add more tips in UG

kexiaowen committed:
Fix whitespace issue

mhq199657 committed:
fix ambiguous wording

kexiaowen committed:
Minor update

kexiaowen committed:
Merge pull request #132 from mhq199657/master

Update manual testing

tanhengyeow committed:
Merge pull request #133 from kexiaowen/v1.5-UGDG

Update User Guide and Developer Guide

tanhengyeow committed:
Updated UGDG & collate final code

tanhengyeow committed:
Merge branch 'master' into v1.5-documentation

tanhengyeow committed:
Merge pull request #134 from tanhengyeow/v1.5-documentation

v1.5-documentation

Ang-YC committed:
Fixed command summary

kexiaowen committed:
Merge pull request #135 from Ang-YC/docs-ug-dg-v3

Fixed command summary

Ang-YC committed:
Shorten down portfolio

Ang-YC committed:
Merge pull request #136 from Ang-YC/docs-ug-dg-v4

Shorten down portfolio

mhq199657 committed:
Add HTML PP

mhq199657 committed:
empty

mhq199657 committed:
Merge remote-tracking branch 'upstream/master' into autopublishing

mhq199657 committed:
autopublishing

mhq199657 committed:
Merge pull request #137 from mhq199657/autopublishing

Autopublishing
